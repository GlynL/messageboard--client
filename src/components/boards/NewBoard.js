import React, { Component } from "react";
import { connect } from "react-redux";
import * as actions from "../../actions/boards";
import requireAuth from "../hocs/requireAuth";

const INITIAL_STATE = { name: "" };

class NewBoard extends Component {
  constructor(props) {
    super(props);
    this.state = INITIAL_STATE;
  }

  handleSubmit = e => {
    e.preventDefault();
    const { history, token } = this.props;
    this.props.addBoard(this.state, token, () => history.push("/"));
  };

  handleChange = e => this.setState({ [e.target.name]: e.target.value });

  render() {
    return (
      <section>
        <header>
          <h1>Add New Board</h1>
        </header>
        <form onSubmit={this.handleSubmit}>
          <label htmlFor="name">Name</label>
          <input
            type="text"
            name="name"
            value={this.state.name}
            onChange={this.handleChange}
          />
          <button>Add Board</button>
        </form>
      </section>
    );
  }
}

const mapStateToProps = state => ({ token: state.auth.authenticated });

export default connect(
  mapStateToProps,
  actions
)(requireAuth(NewBoard));
