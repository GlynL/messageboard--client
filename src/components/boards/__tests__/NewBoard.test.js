import React from "react";
import { shallow } from "enzyme";
import { NewBoard } from "../NewBoard";

let wrapper, addBoard, history, token;

beforeEach(() => {
  addBoard = jest.fn();
  history = { push: jest.fn() };
  token = "test token";
  wrapper = shallow(
    <NewBoard addBoard={addBoard} token={token} history={history} />
  );
});

it("matches snapshot", () => {
  expect(wrapper).toMatchSnapshot();
});

it("handleChange of input", () => {
  wrapper
    .find("input")
    .simulate("change", { target: { name: "name", value: "test" } });
  wrapper.update();
  expect(wrapper.state("name")).toBe("test");
});

it("should handleSubmit", () => {
  const preventDefault = jest.fn();
  wrapper.find("form").simulate("submit", { preventDefault });
  expect(preventDefault).toHaveBeenCalled();
  expect(addBoard).toHaveBeenCalled();
});
