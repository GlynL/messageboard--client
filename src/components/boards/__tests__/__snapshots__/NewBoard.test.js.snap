// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section>
  <header>
    <h1>
      Add New Board
    </h1>
  </header>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="name"
    >
      Name
    </label>
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button>
      Add Board
    </button>
  </form>
</section>
`;
